modules/standard/Set.chpl

```diff
@@ -162,8 +172,13 @@ module Set {
       this.parSafe = parSafe;
       this.complete();
 
-      for x in iterable do
-        _dom.add(x);
+      if !isCopyableType(eltType) then
+        compilerError('Cannot initialize ' + this.type:string + ' from ' +
+                      iterable.type:string + ' because element type ' +
+                      eltType:string + ' is not copyable');
```

Comment:
You can have an iterator returning by value say `owned C` and then all of this should still work... that said it might be an implementation issue to solve later.

Reply:
Oh, because it's constructing new `owned C` as it goes. I'll remove that check, thanks!